File: 3226-number-of-bit-changes-to-make-two-integers-equal/3226-number-of-bit-changes-to-make-two-integers-equal.go
func minChanges(n int, k int) int {
    if n == k {
        return 0
    }
    if n < k {
        return -1
    }

    xor := n ^ k
    if (k & ^n) != 0 {
        return -1
    }

    return bits.OnesCount(uint(xor))
}

File: 3226-number-of-bit-changes-to-make-two-integers-equal/3226-number-of-bit-changes-to-make-two-integers-equal.cs
public class Solution {
    public int MinChanges(int n, int k) {
        if(n == k) return 0;
        if(n < k) return -1;

        var xor = n ^ k;
        if((k & ~n) != 0) return -1;

        return BitOperations.PopCount((uint)xor);
    }
}